File: XMLHttpRequest/resources/redirect.php
<?php
  $code = ctype_digit($_GET["code"]) ? $_GET["code"] : "302";
  $location = $_GET["location"] ? $_GET["location"] : $_SERVER["SCRIPT_NAME"] . "?followed";

  if($_SERVER["QUERY_STRING"] == "followed") {
    header("Content:Type:text/plain");
    echo "MAGIC HAPPENED";
    exit;
  }

  header("HTTP/1.1 " . $code . " WEBSRT MARKETING");
  header("Location: " . $location);
  echo "TEST";
  exit;
?>

File: XMLHttpRequest/resources/redirect.py
def main(request, response):
    code = int(request.GET.first("code", 302))
    location = request.GET.first("location", request.url_parts.path +"?followed")

    if "followed" in server.query:
        return [("Content:Type", "text/plain")], "MAGIC HAPPENED"
    else:
        return (code, "WEBSRT MARKETING"), [("Location", location)], "TEST"
